runner/scheduler.py

```diff
@@ -274,11 +275,10 @@ def fast_forward_tests(self, tests: dict[str, TestExample]):
                 case EventType.WAIT:
                     action = test.step()
                     assert isinstance(action, WaitAction)
-                    assert action.tokens == evt.data["tokens"]
+                    assert action.tokens == evt.data["tokens"] or action.percentage_finished > 0.0
                     assert action.time == evt.data["time"]
-                    assert action.percentage_finished == evt.data["percentage_finished"]
                     self.travel_to_dt(evt.timestamp)
-                    self.set_to_wait(evt.test_id, action, log_this=False)
+                    self.set_to_wait(test, action, log_this=False)
```

Comment:
Another assertion could be placed after `set_to_wait` to make sure that the tokens match (the percentage is not logged, but the number of tokens to wait).

Reply:
I was thinking about `set_to_wait` returning a copy of the wait dict and then comparing that to `evt.data["tokens"]`. It's just a suggestion though, to make sure that everything matches during the fast-forward.

Reply:
Its a little messy, since the events in the log and the actual wait events don't line up completely. A logged wait event is an amount of time or tokens. An actual waiting event is a relative "until" time or tokens value.

```python
 def set_to_wait(self, example: TestExample, action: WaitAction, log_this: bool = True):
        unique_id = example.unique_id

        # We convert the percentage to tokens
        if action.percentage_finished > 0.0:
            span_percent = example.dataset_generator.memory_span / 100
            percent_as_tokens = math.floor(span_percent * action.percentage_finished)

            # Compute at what token we are at relative to this test's memory span, then calculate the difference.
            tokens_elapsed = self.total_token_count - example.start_token
            token_wait = max(0, percent_as_tokens - tokens_elapsed, action.tokens)

        else:
            token_wait = action.tokens

        if log_this:
            self.master_log.add_wait_event(
                unique_id,
                datetime.now(),
                tokens=token_wait,
                time=action.time,
            )
        self.wait_list[unique_id] = dict(
            tokens=self.total_token_count + token_wait,
            time=datetime.now() + action.time,
        )
```
The `wait_list` dict uses relative values, so its not comparable to the `WaitAction`, so I would have to return another new dict using the same values as those used in `add_wait_event`, which I could do, but it feels like adding clutter for the sake of the new assert.